src/future.cpp

```diff
@@ -15,13 +18,25 @@
 #include <mutex>
 #include <thread>
 #include <vector>
+#include <mutex>
+#include <atomic>
```

Comment:
On current merge we get duplicated headers - sort alphabetically to avoid this please.